### PR TITLE
Reports: Added logic to download bhav copy and market activity reports.

### PR DESCRIPTION
**Pull request by jitendravarma**
Added logic to process 
1. Market activity report for a given date and returns raw text (since ma report has lot many inner tables). So, I have let user decide what they want to do with raw content
2. Added logic to download bhav copy zip

Updated Readme.

How was it tested?
Ran locally for multiple use cases.

Breaking changes: No.

### CODE REVIEW

File: nsedt/reports.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        response_type (str, Optional): define the response type panda_df | json . Default json\n
+    Returns:
+        string: string content of the file as right now its not possible 
+        to format the content to json or pandas df
+    Expects: 
+        date to be in format of  "ddmmYY" eg: 30/04/2024 => 300424
+        all other cases will be invalidated
+    """
+    if not validate_date_format(date,format='%d%m%y'):
+        raise ValueError("Please provide date format in 'ddmmYY' format")
```
Comment (pratik141) [inline nsedt/reports.py:28]: can we keep unified date format for all APIs `%d-%m-%Y`

ref: #34

File: nsedt/reports.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        date (str): date for which to download market activity report\n
+        path (str): path to save the bhav copy zip
+    Returns:
+        bool: if the file is save to the local path or not
+    Expects: 
+        date to be in format of  "ddmmYY" eg: 30/04/2024 => 30APR2024
+        all other cases will be invalidated
+    """
+    if not validate_date_format(date, format='%d%b%Y'):
+        raise ValueError("Please provide date format in 'ddMMMYYYY' format")
```
Comment (pratik141) [inline nsedt/reports.py:49]: same date format

**Review by pratik141 — COMMENTED** [review]
Please add test case to validate o/p is comming as expected if possible
please run Pylint also

File: nsedt/reports.py (hunk 2 above)
Comment (jitendravarma) [reply nsedt/reports.py:49]: This is the date format nse expects in the url. Same for the above. Eg: https://nsearchives.nseindia.com/archives/equities/mkt/MA300424.csv and
https://nsearchives.nseindia.com/content/historical/EQUITIES/2024/APR/cm30APR2024bhav.csv.zip

**Comment by jitendravarma** [comment]
> Please add test case to validate o/p is comming as expected if possible please run Pylint also

Updated test cases.

File: nsedt/reports.py (hunk 2 above)
Comment (pratik141) [reply nsedt/reports.py:49]: yes I understand NSE api expect different date format in different places but we in nsedt can try to take unified date format for all APIs and later we can convert it as per nse API requirement.

Comment (jitendravarma) [reply nsedt/reports.py]: Done!

File: tests/test_reports.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+"""
+  Test case for nsedt.reports
+"""
+
+from nsedt.reports import get_bhav_copy_zip, get_market_activity_report
+
+# modify date to the latest date
+MA_ACTIVITY_DATE = "020524"
```
Comment (pratik141) [inline tests/test_reports.py:8]: update date here as well

Comment (jitendravarma) [reply tests/test_reports.py]: Done!

**Comment by jitendravarma** [comment]
Fixed test cases and modified bhav copy download logic now supports df or json.

**Pull request merged**